R/oauth-flow-auth-code.R

```diff
@@ -120,20 +122,26 @@ req_oauth_auth_code <- function(req, client,
 #' @param auth_params List containing additional parameters passed to `oauth_flow_auth_code_url()`
 #' @param token_params List containing additional parameters passed to the
 #'   `token_url`.
-#' @param host_name `r lifecycle::badge("deprecated")` Use `redirect_uri`
-#'   instead.
+#' @param host_name,host_ip,port `r lifecycle::badge("deprecated")`
+#'   Now use `redirect_uri` instead.
 #' @param host_ip IP address for the temporary webserver used to capture the
 #'   authorization code.
-#' @param type Either `desktop` or `web`. Use desktop when running on the
-#'   desktop in an environment where you can redirect the user to `localhost`.
-#'   Use `web` when running in a hosted web environment.
-#' @param port Port to bind the temporary webserver to. Used only when
-#'   `redirect_uri` is `"http(s)://localhost"`. By default, this uses a random
-#'   port. You may need to set it to a fixed port if the API requires that the
-#'   `redirect_uri` specified in the client exactly matches the `redirect_uri`
-#'   generated by this function.
+#' @param port Port to bind the temporary webserver to.
 #' @param redirect_uri URL to redirect back to after authorization is complete.
 #'   Often this must be registered with the API in advance.
+#'
+#'   httr2 supports two forms of redirect. Firstly, you can use a `localhost`
+#'   url (the default), in which case httr2 will set up a temporary webserver
+#'   to listen for the redirect. This works well when you are working on your
+#'   own laptop. In this case, httr2 will automatically append a random port.
```

Comment:
Not everyone has a laptop! 🙈 

```suggestion
#'   own computer. In this case, httr2 will automatically append a random port.
```